telemetry/common/common.go

```diff
@@ -0,0 +1,33 @@
+package common
+
+import (
+	"context"
+
+	"github.com/status-im/status-go/wakuv2/common"
+	wakuv2protocol "github.com/waku-org/go-waku/waku/v2/protocol"
+
+	"github.com/libp2p/go-libp2p/core/protocol"
+
+	"github.com/libp2p/go-libp2p/core/metrics"
+)
+
+type ProtocolStatsMap map[protocol.ID]metrics.Stats
+
+type SentEnvelope struct {
+	Envelope      *wakuv2protocol.Envelope
+	PublishMethod common.PublishMethod
+}
+
+type ErrorSendingEnvelope struct {
+	Error        error
+	SentEnvelope SentEnvelope
+}
+
+type ITelemetryClient interface {
```

Comment:
Should these types live in status-im/telemetry?

Reply:
Good question, they probably could be. I kept them in status-go because they are used in `processAndPushTelemetry` (https://github.com/status-im/status-go/pull/5674/files#diff-6a98f0c7655c1846990bd03c6d7c1feab2b2d96ae9cf14b3bcd3d5c3677af6ffR199) along types like `v2protocol.Envelope` and some other defined in the `client`. 

I'd say we could keep it this way as the next step still is to move the whole client to `telemetry` repo and that will require some more thoughts on the API and/or how to abstract it. 

But also happy move it now if that is the preference

Reply:
nah was just curious! PR looks good already!